jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/server/Dispatcher.java

```diff
@@ -373,11 +382,23 @@ public static Operation chooseOperation(HttpAction action) {
 
         // -- Any other queryString
         // Query string now unexpected.
+
         // Place for an extension point.
         boolean hasParams = request.getParameterMap().size() > 0;
         if ( hasParams ) {
+            // One nasty case:
+            // Bad HTML form (content-type  application/x-www-form-urlencoded), but body is not an HTML form.
+            //  map is one entry, and the key is all of the body,
+            if ( WebContent.contentTypeHTMLForm.equals(request.getContentType()) ) {
+                ServletOps.errorBadRequest("Malformed request: unrecognized HTML form request");
+                return null;
+            }
```

Comment:
I think I got hit by this when I was writing the Vue.js client. The default value was sending JSON, or some other value, not HTML form. I couldn't figure out what was wrong, so had to look at the browser requests in the old UI to understand what the server was expecting.

Reply:
Ah. I came across it with `curl` but from what you say, code defaults POST to a HTML form if there is no content-type given. I guess the common case other that forms is "just send this file, the other end knows what to do" (i.e. ignore the content-type).

Fuseki dispatch is quite complicated and HTML forms are a possibility for SPARQL Update and SPARQL Query so it's not just simply ignoring `application/x-www-form-urlencoded`.